### ISSUE TITLE
UnusedParameter should not warn about class method overrides

### ISSUE THREAD

**Issue by Soullivaneuh** [slevomat/coding-standard#440]
Exemple:

```php
final class ChoiceTypeExtension extends AbstractTypeExtension
{
    public function getExtendedType(): string
    {
        return ChoiceType::class;
    }

    /**
     * @param mixed[] $options
     */
    public function buildView(FormView $view, FormInterface $form, array $options): void
    {
        $view->vars['placeholder_as_choice'] = $options['placeholder_as_choice'];
    }

    public function configureOptions(OptionsResolver $resolver): void
    {
        $resolver
            ->setDefaults([
                'placeholder_as_choice' => false,
            ])
            ->setAllowedTypes('placeholder_as_choice', 'bool')
        ;
    }
}
```

Here, phpcs is warning me about `$form` not being used in `buildView`.

That is right, but I can not do anything about that (can I?) because this method is an override of `AbstractTypeExtension::buildView`, so I can't remove the parameter.

If nothing is possible to fix it, the sniffer should not report that IMO.

What do you think?

**Comment by ondrejmirtes** [comment]
This cannot be solved with PHPCS, you should use `@phpcsSuppress` above the method.

**Comment by EvgenyOrekhov** [comment]
I consider not warning about class method overrides a feature. It forces me to review the code and sometimes that leads to refactoring.
